File: tools-manage/invocationlab-admin/src/main/java/org/javamaster/invocationlab/admin/service/ErdOnlineUserService.java
package org.javamaster.invocationlab.admin.service;

import org.javamaster.invocationlab.admin.model.erd.TokenVo;

import java.util.List;
import java.util.Map;

/**
 * @author yudong
 */
public interface ErdOnlineUserService {
    String findUserName(String code);

    List<Map<String, Object>> findUsers(String name, String code);

    TokenVo login(String userId, String password);

    void refreshToken(String token);

    String logout();

    String changePwd(String newPwd, TokenVo tokenVo);
}

File: tools-manage/invocationlab-admin-21/src/main/kotlin/org/javamaster/invocationlab/admin/service/ErdOnlineUserService.kt
package org.javamaster.invocationlab.admin.service

import org.javamaster.invocationlab.admin.model.erd.TokenVo

/**
 * @author yudong
 */
interface ErdOnlineUserService {
    fun findUserName(code: String): String

    fun findUsers(name: String, code: String): List<Map<String, Any>>

    fun login(userId: String, password: String): TokenVo

    fun refreshToken(token: String)

    fun logout(): String

    fun changePwd(newPwd: String, tokenVo: TokenVo): String
}
